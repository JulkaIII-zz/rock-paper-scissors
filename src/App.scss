$dark-grey: #212121;
$teal: teal;
.App {
  text-align: center;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-between;
  .app-header {
    font-size: calc(10px + 2vmin);
    color: $teal;
  }
  .app-logo {
    animation: App-logo-spin infinite 20s linear;
    height: 35vmin;
  }
  .play-button {
    padding-bottom: 100px;
  }
  main {
    width: 100%;
  }
  .footer-text {
    color: $dark-grey;
  }
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.game-section {
  .userName {
    margin-bottom: 100px;
  }
  .icons-block {
    .choice-icons {
      span:not(:last-of-type) {
        margin-right: 30px;
      }
    }
  }
  .game-button {
    min-height: 35px;
  }
}

.game-result {
  margin-top: 60px;
  color: $teal;
}

.score-table {
  table {
    float: none !important;
    margin: 0 auto;
  }
}
