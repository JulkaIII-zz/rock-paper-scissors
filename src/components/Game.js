import React, { Component } from "react";
import { ICONS_ARRAY, RULES } from "../constants";
import IconSVG from "./IconSVG";
import ScoreTable from "./ScoreTable";
import _ from "lodash";

class Game extends Component {
  constructor() {
    super();
    this.state = {
      userChosenItem: "",
      computerChosenItem: "",
      userClicked: false,
      result: "",
      round: 1,
      userScore: 0,
      computerScore: 0
    };
    this.handleUserClick = this.handleUserClick.bind(this);
    this.handleComputerLogic = this.handleComputerLogic.bind(this);
    this.nextRound = this.nextRound.bind(this);
    this.calculateResult = this.calculateResult.bind(this);
    this.playGame = this.playGame.bind(this);
    this.restartGame = this.restartGame.bind(this);
  }

  playGame(userChosenItem) {
    let { userClicked } = this.state;
    if (userClicked) {
      return;
    }
    this.handleUserClick(userChosenItem);
    let computerChosenItem = this.handleComputerLogic();
    this.calculateResult(userChosenItem, computerChosenItem);
  }

  handleUserClick(userChosenItem) {
    this.setState({ userChosenItem: userChosenItem, userClicked: true });
  }

  handleComputerLogic() {
    const iconNames = ICONS_ARRAY;
    const randomItem = _.sample(iconNames);
    this.setState({ computerChosenItem: randomItem });
    return randomItem;
  }

  calculateResult(userChosenItem, computerChosenItem) {
    if (userChosenItem === computerChosenItem) {
      this.setState({ result: "Draw" });
      return;
    } else if (RULES[userChosenItem] === computerChosenItem) {
      this.setState({ result: "You win", userScore: this.state.userScore + 1 });
      return;
    }
    return this.setState({
      result: "You lost",
      computerScore: this.state.computerScore + 1
    });
  }

  nextRound() {
    this.setState({
      userChosenItem: "",
      computerChosenItem: "",
      userClicked: false,
      result: "",
      round: this.state.round + 1
    });
  }

  restartGame() {
    this.setState({
      userChosenItem: "",
      computerChosenItem: "",
      userClicked: false,
      result: "",
      round: 1,
      userScore: 0,
      computerScore: 0
    });
  }

  render() {
    let {
      userChosenItem,
      computerChosenItem,
      userClicked,
      result,
      round,
      userScore,
      computerScore
    } = this.state;
    const iconNames = ICONS_ARRAY;
    const scores = [userScore, computerScore];
    let userChoiceItems = _.map(iconNames, (iconName, index) => {
      if (userChosenItem === "" || userChosenItem === iconName) {
        return (
          <div
            className="col s12 m4"
            key={index}
            onClick={() => this.playGame(iconName)}
          >
            <IconSVG name={iconName} />
          </div>
        );
      } else {
        return <div className="col s12 m4" key={index} />;
      }
    });

    let computerChoiceItems = _.map(iconNames, (iconName, index) => {
      if (computerChosenItem === "" || computerChosenItem === iconName) {
        return (
          <div className="col s12 m4" key={index}>
            <IconSVG name={iconName} />
          </div>
        );
      } else {
        return <div className="col s12 m4" key={index} />;
      }
    });

    return (
      <section className="game-section">
        <div>
          <div className="row">
            <div className="col s5 icons-block">
              <h4 className="col s12 user-name">You</h4>
              <div className="choice-icons">{userChoiceItems}</div>
            </div>
            <div className="col s2">
              <h5 className="col s12 teal accent-2 round">Round {round}</h5>
              <h4 className="col s12 game-result">{result}</h4>
            </div>
            <div className="col s5">
              <h4 className="col s12 computer-name">Computer</h4>
              <div className="choice-icons">{computerChoiceItems}</div>
            </div>
          </div>
          <div className="game-button">
            {userClicked && (
              <a
                className="waves-effect waves-light btn-small next-round-btn"
                onClick={this.nextRound}
              >
                <i className="material-icons right">navigate_next</i>Next Round
              </a>
            )}
          </div>
          <ScoreTable scores={scores} />
          <div className="game-button">
            {(round > 1 || userClicked) && (
              <a
                className="waves-effect waves-light btn-small restart-btn"
                onClick={this.restartGame}
              >
                <i className="material-icons right">replay</i>Restart game
              </a>
            )}
          </div>
        </div>
      </section>
    );
  }
}

export default Game;
