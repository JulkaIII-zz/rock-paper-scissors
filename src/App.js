import React, { Component } from "react";
import logo from "./icons/logo.svg";
import Game from "./components/Game";
import "./App.scss";
import "materialize-css/dist/css/materialize.min.css";

class App extends Component {
  constructor() {
    super();
    this.state = { renderComponent: false };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    console.log("APP event");
    this.setState({ renderComponent: !this.state.renderComponent });
  }

  render() {
    const { renderComponent } = this.state;
    return (
      <div className="App">
        <header className="app-header">
          <h1>Rock Paper Scissors</h1>
        </header>
        <main>
          {renderComponent ? (
            <Game />
          ) : (
            <img src={logo} className="app-logo" alt="logo" />
          )}
        </main>
        <footer>
          <div className="play-button" onClick={this.handleClick}>
            {renderComponent ? (
              <a className="waves-effect waves-light btn-large">
                <i className="material-icons right">exit_to_app</i>
                quit
              </a>
            ) : (
              <a className="waves-effect waves-light btn-large">
                <i className="material-icons right">play_arrow</i>
                play
              </a>
            )}
          </div>
          <span className="footer-text">2018 &copy;</span>
        </footer>
      </div>
    );
  }
}

export default App;
