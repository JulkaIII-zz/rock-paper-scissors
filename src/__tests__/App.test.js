import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, configure } from "enzyme";
import App from "../App";
import Game from "../components/Game";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("App component", function() {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it("should render App successfully", () => {
    const appElement = wrapper.find(".App");
    expect(appElement).toMatchSnapshot();
  });

  it("should have a proper class name", () => {
    const appElement = wrapper.hasClass("App");
    expect(appElement).toBeTruthy();
  });

  it("should have a Play button", () => {
    const playButton = wrapper.find(".play-button");
    expect(playButton).toBeTruthy();
  });

  it("should have an onClick handler on Play button", () => {
    const handler = wrapper.find(".play-button").props().onClick;
    expect(handler).toBeTruthy();
  });

  it("should have App state", () => {
    const appState = wrapper.state().renderComponent;
    expect(appState).toBe(false);
  });

  it("should have a handleClick method", () => {
    const handler = wrapper.instance().handleClick;
    expect(handler).toBeTruthy();
  });

  it("should click on a Play button and change App state", () => {
    wrapper.find(".play-button").simulate("click");
    const appState = wrapper.state().renderComponent;
    expect(appState).toBe(true);
  });

  it("should include a Game component after clicking Play button", () => {
    wrapper.find(".play-button").simulate("click");
    const gameComponent = wrapper.find(Game).length;
    expect(gameComponent).toEqual(1);
  });

  it("should change a Play button to a Quit button after clicking Play button", () => {
    wrapper.find(".play-button").simulate("click");
    const quitButtonText = wrapper
      .find(".play-button")
      .childAt(0)
      .text();
    expect(quitButtonText).toContain("quit");
  });
});
