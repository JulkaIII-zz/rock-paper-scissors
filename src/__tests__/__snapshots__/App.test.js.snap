// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render App successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): ShallowWrapper {
    Symbol(enzyme.__root__): [Circular],
    Symbol(enzyme.__unrendered__): <App />,
    Symbol(enzyme.__renderer__): Object {
      "batchedUpdates": [Function],
      "getNode": [Function],
      "render": [Function],
      "simulateError": [Function],
      "simulateEvent": [Function],
      "unmount": [Function],
    },
    Symbol(enzyme.__node__): Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="app-header"
          >
            <h1>
              Rock Paper Scissors
            </h1>
          </header>,
          <main>
            <img
              alt="logo"
              className="app-logo"
              src="logo.svg"
            />
          </main>,
          <footer>
            <div
              className="play-button"
              onClick={[Function]}
            >
              <a
                className="waves-effect waves-light btn-large"
              >
                <i
                  className="material-icons right"
                >
                  play_arrow
                </i>
                play
              </a>
            </div>
            <span
              className="footer-text"
            >
              2018 ©
            </span>
          </footer>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Rock Paper Scissors
            </h1>,
            "className": "app-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rock Paper Scissors",
            },
            "ref": null,
            "rendered": "Rock Paper Scissors",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo"
              className="app-logo"
              src="logo.svg"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "app-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="play-button"
                onClick={[Function]}
              >
                <a
                  className="waves-effect waves-light btn-large"
                >
                  <i
                    className="material-icons right"
                  >
                    play_arrow
                  </i>
                  play
                </a>
              </div>,
              <span
                className="footer-text"
              >
                2018 ©
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="waves-effect waves-light btn-large"
                >
                  <i
                    className="material-icons right"
                  >
                    play_arrow
                  </i>
                  play
                </a>,
                "className": "play-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="material-icons right"
                    >
                      play_arrow
                    </i>,
                    "play",
                  ],
                  "className": "waves-effect waves-light btn-large",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "play_arrow",
                      "className": "material-icons right",
                    },
                    "ref": null,
                    "rendered": "play_arrow",
                    "type": "i",
                  },
                  "play",
                ],
                "type": "a",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2018 ©",
                "className": "footer-text",
              },
              "ref": null,
              "rendered": "2018 ©",
              "type": "span",
            },
          ],
          "type": "footer",
        },
      ],
      "type": "div",
    },
    Symbol(enzyme.__nodes__): Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <header
              className="app-header"
            >
              <h1>
                Rock Paper Scissors
              </h1>
            </header>,
            <main>
              <img
                alt="logo"
                className="app-logo"
                src="logo.svg"
              />
            </main>,
            <footer>
              <div
                className="play-button"
                onClick={[Function]}
              >
                <a
                  className="waves-effect waves-light btn-large"
                >
                  <i
                    className="material-icons right"
                  >
                    play_arrow
                  </i>
                  play
                </a>
              </div>
              <span
                className="footer-text"
              >
                2018 ©
              </span>
            </footer>,
          ],
          "className": "App",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h1>
                Rock Paper Scissors
              </h1>,
              "className": "app-header",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rock Paper Scissors",
              },
              "ref": null,
              "rendered": "Rock Paper Scissors",
              "type": "h1",
            },
            "type": "header",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="logo"
                className="app-logo"
                src="logo.svg"
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "logo",
                "className": "app-logo",
                "src": "logo.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "main",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="play-button"
                  onClick={[Function]}
                >
                  <a
                    className="waves-effect waves-light btn-large"
                  >
                    <i
                      className="material-icons right"
                    >
                      play_arrow
                    </i>
                    play
                  </a>
                </div>,
                <span
                  className="footer-text"
                >
                  2018 ©
                </span>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <a
                    className="waves-effect waves-light btn-large"
                  >
                    <i
                      className="material-icons right"
                    >
                      play_arrow
                    </i>
                    play
                  </a>,
                  "className": "play-button",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="material-icons right"
                      >
                        play_arrow
                      </i>,
                      "play",
                    ],
                    "className": "waves-effect waves-light btn-large",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "play_arrow",
                        "className": "material-icons right",
                      },
                      "ref": null,
                      "rendered": "play_arrow",
                      "type": "i",
                    },
                    "play",
                  ],
                  "type": "a",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "2018 ©",
                  "className": "footer-text",
                },
                "ref": null,
                "rendered": "2018 ©",
                "type": "span",
              },
            ],
            "type": "footer",
          },
        ],
        "type": "div",
      },
    ],
    Symbol(enzyme.__options__): Object {
      "adapter": ReactSixteenAdapter {
        "options": Object {
          "enableComponentDidUpdateOnSetState": true,
          "lifecycles": Object {
            "componentDidUpdate": Object {
              "onSetState": true,
            },
            "getDerivedStateFromProps": true,
            "getSnapshotBeforeUpdate": true,
            "setState": Object {
              "skipsComponentDidUpdateOnNullish": true,
            },
          },
        },
      },
    },
  },
  Symbol(enzyme.__unrendered__): null,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="app-header"
        >
          <h1>
            Rock Paper Scissors
          </h1>
        </header>,
        <main>
          <img
            alt="logo"
            className="app-logo"
            src="logo.svg"
          />
        </main>,
        <footer>
          <div
            className="play-button"
            onClick={[Function]}
          >
            <a
              className="waves-effect waves-light btn-large"
            >
              <i
                className="material-icons right"
              >
                play_arrow
              </i>
              play
            </a>
          </div>
          <span
            className="footer-text"
          >
            2018 ©
          </span>
        </footer>,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <h1>
            Rock Paper Scissors
          </h1>,
          "className": "app-header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Rock Paper Scissors",
          },
          "ref": null,
          "rendered": "Rock Paper Scissors",
          "type": "h1",
        },
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="logo"
            className="app-logo"
            src="logo.svg"
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "logo",
            "className": "app-logo",
            "src": "logo.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "main",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="play-button"
              onClick={[Function]}
            >
              <a
                className="waves-effect waves-light btn-large"
              >
                <i
                  className="material-icons right"
                >
                  play_arrow
                </i>
                play
              </a>
            </div>,
            <span
              className="footer-text"
            >
              2018 ©
            </span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="waves-effect waves-light btn-large"
              >
                <i
                  className="material-icons right"
                >
                  play_arrow
                </i>
                play
              </a>,
              "className": "play-button",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <i
                    className="material-icons right"
                  >
                    play_arrow
                  </i>,
                  "play",
                ],
                "className": "waves-effect waves-light btn-large",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "play_arrow",
                    "className": "material-icons right",
                  },
                  "ref": null,
                  "rendered": "play_arrow",
                  "type": "i",
                },
                "play",
              ],
              "type": "a",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2018 ©",
              "className": "footer-text",
            },
            "ref": null,
            "rendered": "2018 ©",
            "type": "span",
          },
        ],
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="app-header"
          >
            <h1>
              Rock Paper Scissors
            </h1>
          </header>,
          <main>
            <img
              alt="logo"
              className="app-logo"
              src="logo.svg"
            />
          </main>,
          <footer>
            <div
              className="play-button"
              onClick={[Function]}
            >
              <a
                className="waves-effect waves-light btn-large"
              >
                <i
                  className="material-icons right"
                >
                  play_arrow
                </i>
                play
              </a>
            </div>
            <span
              className="footer-text"
            >
              2018 ©
            </span>
          </footer>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Rock Paper Scissors
            </h1>,
            "className": "app-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rock Paper Scissors",
            },
            "ref": null,
            "rendered": "Rock Paper Scissors",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo"
              className="app-logo"
              src="logo.svg"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "app-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="play-button"
                onClick={[Function]}
              >
                <a
                  className="waves-effect waves-light btn-large"
                >
                  <i
                    className="material-icons right"
                  >
                    play_arrow
                  </i>
                  play
                </a>
              </div>,
              <span
                className="footer-text"
              >
                2018 ©
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="waves-effect waves-light btn-large"
                >
                  <i
                    className="material-icons right"
                  >
                    play_arrow
                  </i>
                  play
                </a>,
                "className": "play-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="material-icons right"
                    >
                      play_arrow
                    </i>,
                    "play",
                  ],
                  "className": "waves-effect waves-light btn-large",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "play_arrow",
                      "className": "material-icons right",
                    },
                    "ref": null,
                    "rendered": "play_arrow",
                    "type": "i",
                  },
                  "play",
                ],
                "type": "a",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2018 ©",
                "className": "footer-text",
              },
              "ref": null,
              "rendered": "2018 ©",
              "type": "span",
            },
          ],
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__rootNodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="app-header"
          >
            <h1>
              Rock Paper Scissors
            </h1>
          </header>,
          <main>
            <img
              alt="logo"
              className="app-logo"
              src="logo.svg"
            />
          </main>,
          <footer>
            <div
              className="play-button"
              onClick={[Function]}
            >
              <a
                className="waves-effect waves-light btn-large"
              >
                <i
                  className="material-icons right"
                >
                  play_arrow
                </i>
                play
              </a>
            </div>
            <span
              className="footer-text"
            >
              2018 ©
            </span>
          </footer>,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h1>
              Rock Paper Scissors
            </h1>,
            "className": "app-header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rock Paper Scissors",
            },
            "ref": null,
            "rendered": "Rock Paper Scissors",
            "type": "h1",
          },
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="logo"
              className="app-logo"
              src="logo.svg"
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "logo",
              "className": "app-logo",
              "src": "logo.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "main",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="play-button"
                onClick={[Function]}
              >
                <a
                  className="waves-effect waves-light btn-large"
                >
                  <i
                    className="material-icons right"
                  >
                    play_arrow
                  </i>
                  play
                </a>
              </div>,
              <span
                className="footer-text"
              >
                2018 ©
              </span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="waves-effect waves-light btn-large"
                >
                  <i
                    className="material-icons right"
                  >
                    play_arrow
                  </i>
                  play
                </a>,
                "className": "play-button",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <i
                      className="material-icons right"
                    >
                      play_arrow
                    </i>,
                    "play",
                  ],
                  "className": "waves-effect waves-light btn-large",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "play_arrow",
                      "className": "material-icons right",
                    },
                    "ref": null,
                    "rendered": "play_arrow",
                    "type": "i",
                  },
                  "play",
                ],
                "type": "a",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "2018 ©",
                "className": "footer-text",
              },
              "ref": null,
              "rendered": "2018 ©",
              "type": "span",
            },
          ],
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
}
`;
